persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageInsertExtension.kt

```diff
@@ -303,6 +316,29 @@ internal class MessageInsertExtensionImpl(
         }
     }
 
+    private fun insertCompositeMessage(
+        content: MessageEntityContent.Composite,
+        message: MessageEntity
+    ) {
+        content.text?.let { text ->
+            messagesQueries.insertMessageTextContent(
+                message_id = message.id,
+                conversation_id = message.conversationId,
+                text_body = text.messageBody,
+                quoted_message_id = text.quotedMessageId,
+                is_quote_verified = text.isQuoteVerified
+            )
+        }
+        content.buttonList.forEach { button ->
+            buttonContentQueries.insertButton(
+                message_id = message.id,
+                conversation_id = message.conversationId,
+                id = button.id,
+                text = button.text
+            )
+        }
```

Comment:
I am not aware of a way to insert multiple rows dynamically in SQL, and since all of this is in one transaction, there should be no performance issues

Reply:
I trust u on that as I am still SQL newbie